modules/home-environment.nix

```diff
@@ -577,25 +577,38 @@ in
       if config.submoduleSupport.externalPackageInstall
       then
         ''
-          if nix-env -q | grep '^home-manager-path$'; then
+         if [[ -e "$genProfilePath"/manifest.json ]] ; then
+           nix profile list | { grep 'home-manager-path$' || test $? = 1; } | awk -F ' ' '{ print $4 }' | xargs -t $DRY_RUN_CMD nix profile remove $VERBOSE_ARG
+         else
+           if nix-env -q | grep '^home-manager-path$'; then
             $DRY_RUN_CMD nix-env -e home-manager-path
-          fi
+           fi
+         fi
         ''
       else
         ''
-          if ! $DRY_RUN_CMD nix-env -i ${cfg.path} ; then
+          if [[ -e "$genProfilePath"/manifest.json ]] ; then
+            INSTALL_CMD="nix profile install"
+            LIST_CMD="nix profile list"
+            REMOVE_CMD_SYNTAX='nix profile remove {number | store path}'
+          else
+            INSTALL_CMD="nix-env -i"
+            LIST_CMD="nix-env -q"
+            REMOVE_CMD_SYNTAX='nix-env -e {package name}'
+          fi
+          if ! $DRY_RUN_CMD $INSTALL_CMD ${cfg.path} ; then
```

Comment:
- `nix profile` expands to nix `profile --profile /nix/var/nix/profiles/per-user/$user`
- `"$genProfilePath"/manifest.json` expands to `/nix/var/nix/profiles/per-user/$user/home-manager/manifest.json`

Which one is correct?

---

It also strikes me as volatile to rely on the environment to determine the profile path, in case the environment user doesn't match the intended target user. Such env user could be a deployment user.